### DISCUSSION TITLE
Problem with low level library

### DISCUSSION THREAD

**Discussion by Budgie2** [ImageMagick/ImageMagick#4199] [Help]
### Prerequisites

- [ ] I have written a descriptive issue title
- [ ] I have verified that I am using the latest version of ImageMagick
- [ ] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### System Configuration

- ImageMagick version:
- Environment (Operating system, version and so on):
- Additional information:

**Comment by Budgie2** [comment]
I am running openSUSE Tumbleweed which was used to build:-
`Version: ImageMagick 7.0.10-38 Q16 x86_64 2020-11-14 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib fpx jpeg x zlib`

the delegate fpx (libfpx) I also built from the master and the version is:-
`2017-07-04  1.3.1-9 nros
	* Patches to fix  CVE-2017-12921 and CVE-2017-12925 and possibly
    CVE-2017-12920.`

I am using this application to work with digiKam as an add-on which is a recognised add-on.
My problem is that if I have .fpx images in my album, then digiKam crashes and exits.  I have tried with only one .fpx image and this works, ie the thumbnail and full image open as they should but adding a second image to the album gave unreliable results and with a third image digiKam crashes every time.  
I first raised a bug with digiKam but following sending backtrace for that bug I received the following message

> It crash in the flashpix library used by ImageMagick.

It sound like the problem is not in digiKam as well, but in low level library
used to decode image.

Can you process this fpx image with ImageMagick, for ex using "display" command
line tool ?

If ImageMagick crash also, please report this problem as UPSTREAM to
ImageMagick team. 

Following this request I checked and can confirm that IM does not crash.  I tried using it to convert the .fpx images to .jpg and IM ran perfectly but I have been asked to report this as an UPSTREAM bug, hence this post.

If you need more info please let me know.  I am not a coder so please forgive me if I need help.
Regards,
Budge

**Comment by Budgie2** [comment]
Sorry I have misused the quotes tool here.  the quote from digiKam is 



    It crash in the flashpix library used by ImageMagick.

It sound like the problem is not in digiKam as well, but in low level library
used to decode image.

Can you process this fpx image with ImageMagick, for ex using "display" command
line tool ?

If ImageMagick crash also, please report this problem as UPSTREAM to
ImageMagick team.

**Comment by snibgo** [comment]
>  I tried using it to convert the .fpx images to .jpg and IM ran perfectly ...

I doubt if we can help. Perhaps you can provide a link to a FPX file that gives digiKam problems, and we can try converting it with different versions of IM.

**Comment by Budgie2** [comment]
Hi snibgo,  I have over 1000 files from an old Kodak and I no longer have the camera.  They run perfectly on IM but not in digikam.  I have no idea how digikam works with IM and libfpx but it is possible I have the configuration of my system wrong.  I tried to attach a few fpx images so you could reproduce the problem but this tells me the file type is not supported.  What is best way to get the images to you please?

**Comment by fmw42** [comment]
I suggest you zip the group of files and attach the zip file.

**Comment by Budgie2** [comment]
Of course.  Try the attached.  Should hold 5 .fpx files
[Templates.zip](https://github.com/ImageMagick/ImageMagick/files/5639333/Templates.zip)

**Comment by urban-warrior** [comment]
We downloaded the latest libfpx library from https://github.com/ImageMagick/libfpx.git on our Fedora host.  We could read/view all of the FlashPix images you provided without complaint.  We added a memory debugger and no memory corruption was found.

**Comment by Budgie2** [comment]
Hi and thanks for this.  I did not have any trouble from IM but only when using digiKam.  Not being a coder I do not know how these two applications are supposed to work together but the problem falls between the two and hence my question.  Have you tried running digikam on these fpx images?

**Comment by urban-warrior** [comment]
This forum concerns ImageMagick only.  We provide a [discussion]( https://github.com/ImageMagick/ImageMagick/discussions?discussions_q=category%3A%22Digital+Image+Processing%22) category for general image processing topics.

**Comment by fmw42** [comment]
Have you tried contacting the digiKam developers or forum about this providing your images to them?

**Comment by Budgie2** [comment]
On 06/12/2020 17:10, Fred Weinhaus wrote:
> Have you tried contacting the digiKam developers or forum about this providing your images to them?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/ImageMagick/ImageMagick/issues/2903#issuecomment-739532665>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AALT4X6EX3F2MRIMSDAF3TTSTO3H5ANCNFSM4UKQB24A>.
> 

Hi,
I have been trying to pick up where I left off and am unable to log into the forum or even use the lost pw form as I do not get a reply.  No idea why not but only link I have now is bug report.  Please can somebody help me here.  
Regards,
Budge
